### PR TITLE
[86bydf2fy][dropdown-menu] a11y improvements

### PR DESCRIPTION
**Pull request by ilyabrower**
## Motivation and Context
I've changed focus (a11y) behavior for dropdown-menu.
- Switched `aria-activedescendant` to `row-in-tabindex`
- Added possibility to add some actions for menu item (as submenu)
- Refactored nesting menu
- Refactored examples

## How has this been tested?
Manually

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [X] Nice improve.

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly or it's not required.
- [ ] Unit tests are not broken.
- [ ] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [ ] I have added new unit tests on added of fixed functionality.

### CODE REVIEW

**Review by j-mnizhek — APPROVED** [review]
Updated a11y documentation for DropdownMenu.

@sheila-semrush Sheila, please take a look at the updated accessibility documenation for DropdownMenu 🙏

**Comment by j-mnizhek** [comment]
I have a couple of questions:
1. Should we add documentation for `DropdownMenu.Item.Content` and `DropdownMenu.Group` to the API page?

2. Can we center ChevronRight icon a little bit? Now it looks a little bit higher than it was.

**Current:**
image

**Before:**
image

**Comment by j-mnizhek** [comment]
I've just added:

1. Styles for Button with S size, so we can use such a small Button for the cases when we need these kind of an interactive addons inside some other components when there is not much space.
image

2. New token for S sized controls with 20px height.

_Now I'm updating some illustrations for the design guide, will push them a little bit later._

**Review by sheila-semrush — CHANGES_REQUESTED** [review]
Left a few comments

File: website/docs/components/dropdown-menu/dropdown-menu-a11y.md [hunk 1]
```diff
@@ -7,32 +7,41 @@ tabs: Design('dropdown-menu'), A11y('dropdown-menu-a11y'), API('dropdown-menu-ap
...
+| `DropdownMenu.Menu`                              | `menu`                       | The `menu` role is a type of composite widget that offers a list of choices to the user.                                                         |
+|                                                  | `aria-labelledby="IDREF"`    | Identifies the title for the menu.                                                                                                               |
+| `DropdownMenu.Item`, `DropdownMenu.Item.Content` | `menuitem`                   | Indicates the element is an option in a set of choices contained by a `menu`.                                                                    |
+|                                                  | `aria-describedby="IDREF"`   | Gives the item an accessible description by referring to the item hint that describes the primary message or purpose of the item.                |
+| `DropdownMenu.Item.Hint`                         | `aria-hidden="true"`         | Hides element from assistive technologies.                                                                                                       |
```
[the remark this answers is not in the document]
Comment (j-mnizhek) [reply website/docs/components/dropdown-menu/dropdown-menu-a11y.md:39]: I see that Iliia has updated documentation in the API doc, finally. There were some old names, I'll fix to the new ones of course ❤️

File: website/docs/components/dropdown-menu/dropdown-menu.md
[the remark this answers is not in the document]
Comment (j-mnizhek) [reply website/docs/components/dropdown-menu/dropdown-menu.md:1]: Yesss! Was thinking about it too, actually ❤️

File: website/docs/components/dropdown-menu/dropdown-menu-a11y.md [hunk 2]
```diff
@@ -7,32 +7,41 @@ tabs: Design('dropdown-menu'), A11y('dropdown-menu-a11y'), API('dropdown-menu-ap
...
+| `DropdownMenu.ItemTitle`                         | `aria-hidden="true"`         | Hides element from assistive technologies.                                                                                                       |
+| `DropdownMenu.Item.Content`                      | `aria-haspopup="dialog"`     | Indicates that the element triggers a dialog.                                                                                                    |
```
[the remark this answers is not in the document]
Comment (j-mnizhek) [reply website/docs/components/dropdown-menu/dropdown-menu-a11y.md:41]: These components are not only for the nested menus, so I guess it's okay to leave them without this additional comment.

[the remark this answers is not in the document]
Comment (sheila-semrush) [reply website/docs/components/dropdown-menu/dropdown-menu-a11y.md:41]: but the descriptions relate to the properties, not to the components. Current description can make an impression that the property is always there

**Review by sheila-semrush — CHANGES_REQUESTED** [review]
@ilyabrower I've made 2 commits, 1st with some cosmetic documentation changes, and 2nd replacing most triggers with simple buttons. I think they're more appropriate for our examples given that we don't want to show changing content in the triggers. (Julia agreed)

There's one bug left, with menu not closing on ESC when a list item with nested controls is selected.

**Comment by ilyabrower** [comment]
> @ilyabrower I've made 2 commits, 1st with some cosmetic documentation changes, and 2nd replacing most triggers with simple buttons. I think they're more appropriate for our examples given that we don't want to show changing content in the triggers. (Julia agreed)
> 
> There's one bug left, with menu not closing on ESC when a list item with nested controls is selected.

I fixed bug with ESC

**Review by sheila-semrush — APPROVED** [review]
Works great!

**Review by msereniti — APPROVED** [review]

**Pull request merged**